apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-content.js

```diff
@@ -3,14 +3,44 @@ import { ExternalLink } from '@wordpress/components';
 import { createInterpolateElement } from '@wordpress/element';
 import { __, hasTranslation } from '@wordpress/i18n';
 
-const addBlock = 'https://s0.wp.com/i/editor-welcome-tour/slide-add-block.gif';
-const allBlocks = 'https://s0.wp.com/i/editor-welcome-tour/slide-all-blocks.gif';
-const finish = 'https://s0.wp.com/i/editor-welcome-tour/slide-finish.png';
-const makeBold = 'https://s0.wp.com/i/editor-welcome-tour/slide-make-bold.gif';
-const moreOptions = 'https://s0.wp.com/i/editor-welcome-tour/slide-more-options.gif';
-const moveBlock = 'https://s0.wp.com/i/editor-welcome-tour/slide-move-block.gif';
-const undo = 'https://s0.wp.com/i/editor-welcome-tour/slide-undo.gif';
-const welcome = 'https://s0.wp.com/i/editor-welcome-tour/slide-welcome.png';
+/**
+ * This function returns a collection of NUX Tour slide assets
+ *
+ * @param {string} key the lookup string to fetch the proper asset set
+ * @returns a set of <WelcomeTourCard /> assets
+ */
+
+function getTourAssets( key ) {
+	if ( ! key ) {
+		return;
+	}
```

Comment:
> I think this is not necessary either. `tourAssets[ undefined ]` will return `undefined`

Yep, that's correct but the function without the check will be significantly slower when called without arguments.

<img width="845" alt="Screenshot 2021-09-24 at 11 24 31" src="https://user-images.githubusercontent.com/12104589/134651681-095ebb5e-23a7-4a54-bf73-4a5db0067008.png">

Link to the benchmark test: https://jsben.ch/rbKVv

On the other hand, calling the function with arguments doesn't make much difference when it comes to speed. Without the check, the function is slightly faster.

<img width="845" alt="Screenshot 2021-09-24 at 11 27 20" src="https://user-images.githubusercontent.com/12104589/134652079-217fc56a-f069-4293-8b92-7a5d14a30f46.png">

Link to the benchmark test: https://jsben.ch/D2xIW

Since we (currently) always call it with arguments (the function is only used in `getTourContent()`, where arguments are always provided), it makes sense to remove the check, which I did in af32ab174a79650981b04eb7c1154433eb36c49c

Reply:
Very nice. Always appreciate a good thought on optimization. As long as there's a pragmatic focus - to not feel premature and unnecessary.